### ISSUE TITLE
firewall rich rules

### ISSUE THREAD

**Issue by gueniewie** [firewalld/firewalld#796]
in centos 8.3 the new set for freeipa, freeipa-4.xml is not working in rich rule setup

**Comment by erig0** [comment]
This is a low quality report. Please add more information. What is not working? What do you expect?

**Comment by gueniewie** [comment]
helo,
when I install freeipa-4.xml it is working
firewall-cmd --permanent --zone=external --add-service=freeipa-4

but when i have to install freeipa-4.xml in a "rich rule" the freeipa-4.xml is not working?
firewall-cmd --permanent --zone=external --add-rich-rule='rule family="ipv4" source address="xxx.xxx.xxx.xxx/28" service name="freeipa-4" log prefix="freeipa-4" level="info" limit value="1/m" accept'

the firewall is blocked
with the "old" freeipa-ldaps, freeipa-ldap, freeipa-replication.xml allis working

**Comment by erig0** [comment]
> helo,
> when I install freeipa-4.xml it is working
> firewall-cmd --permanent --zone=external --add-service=freeipa-4
> 
> but when i have to install freeipa-4.xml in a "rich rule" the freeipa-4.xml is not working?
> firewall-cmd --permanent --zone=external --add-rich-rule='rule family="ipv4" source address="xxx.xxx.xxx.xxx/28" service name="freeipa-4" log prefix="freeipa-4" level="info" limit value="1/m" accept'

The `limit` stanza makes it so only one new connection per minute is accepted. Perhaps you're exceeding that limit.

**Comment by gueniewie** [comment]
the old freeipa-ldap freeipa-ldaps .........are working in the "rich rules" but the new freeipa-4 I mean is loading but not working it is not opening the ports

I can also drop the limit stanza no changes, but I have read this is only for logs ?

**Comment by erig0** [comment]
> the old freeipa-ldap freeipa-ldaps .........are working in the "rich rules" but the new freeipa-4 I mean is loading but not working it is not opening the ports
> 
> I can also drop the limit stanza no changes, but I have read this is only for logs ?

Not true. Limit applies to log and actions (drop, accept). I suspect the `limit` is your issues. If you want to limit the logging, then you should create two rich rules: one to log with limit, and one to accept.

https://github.com/firewalld/firewalld/blob/44ffde4913af23790891f1b8f891059f651a2f1c/doc/xml/firewalld.richlanguage.xml#L351-L352
